lib/Sema/TypeCheckType.cpp

```diff
@@ -5343,9 +5343,15 @@ void TypeChecker::checkExistentialTypes(Decl *decl) {
   } else if (auto *macroDecl = dyn_cast<MacroDecl>(decl)) {
     checkExistentialTypes(ctx, macroDecl->getGenericParams());
     checkExistentialTypes(ctx, macroDecl->getTrailingWhereClause());
+  } else if (auto *macroExpansionDecl = dyn_cast<MacroExpansionDecl>(decl)) {
+    ExistentialTypeVisitor visitor(ctx, /*checkStatements=*/false);
```

Comment:
Should `checkStatements` be `false` here? What about a freestanding declaration macro with a closure argument?

Reply:
I wondered about that. When we set `checkStatements` `walkToDeclPre` will always skip. The following test below passes:
```swift
#anonymousTypes { () -> String in
  // expected-error @+1 {{use of protocol 'Equatable' as a type must be written 'any Equatable'}}
  _ = 0 as Equatable
  return "foo"
}
``
Do you know an example that might fail without `checkStatements`?